conda/env/specs/yaml_file.py

```diff
@@ -66,5 +59,5 @@ def environment(self) -> EnvironmentYaml:
     @property
     def env(self) -> Environment:
         if not self._environment:
-            self.can_handle()
+            self._environment = env.from_file(self.filename)
```

Comment:
now we can load the environment when the user asks for the environment to be loaded.